lib/identity-document.js

```diff
@@ -0,0 +1,91 @@
+const {spawn} = require('child_process');
+const temporary = require('temporary');
+const fs = require('fs');
+
+const AWS_PUBLIC_CERTIFICATE =
+`-----BEGIN CERTIFICATE-----
+MIIC7TCCAq0CCQCWukjZ5V4aZzAJBgcqhkjOOAQDMFwxCzAJBgNVBAYTAlVTMRkw
+FwYDVQQIExBXYXNoaW5ndG9uIFN0YXRlMRAwDgYDVQQHEwdTZWF0dGxlMSAwHgYD
+VQQKExdBbWF6b24gV2ViIFNlcnZpY2VzIExMQzAeFw0xMjAxMDUxMjU2MTJaFw0z
+ODAxMDUxMjU2MTJaMFwxCzAJBgNVBAYTAlVTMRkwFwYDVQQIExBXYXNoaW5ndG9u
+IFN0YXRlMRAwDgYDVQQHEwdTZWF0dGxlMSAwHgYDVQQKExdBbWF6b24gV2ViIFNl
+cnZpY2VzIExMQzCCAbcwggEsBgcqhkjOOAQBMIIBHwKBgQCjkvcS2bb1VQ4yt/5e
+ih5OO6kK/n1Lzllr7D8ZwtQP8fOEpp5E2ng+D6Ud1Z1gYipr58Kj3nssSNpI6bX3
+VyIQzK7wLclnd/YozqNNmgIyZecN7EglK9ITHJLP+x8FtUpt3QbyYXJdmVMegN6P
+hviYt5JH/nYl4hh3Pa1HJdskgQIVALVJ3ER11+Ko4tP6nwvHwh6+ERYRAoGBAI1j
+k+tkqMVHuAFcvAGKocTgsjJem6/5qomzJuKDmbJNu9Qxw3rAotXau8Qe+MBcJl/U
+hhy1KHVpCGl9fueQ2s6IL0CaO/buycU1CiYQk40KNHCcHfNiZbdlx1E9rpUp7bnF
+lRa2v1ntMX3caRVDdbtPEWmdxSCYsYFDk4mZrOLBA4GEAAKBgEbmeve5f8LIE/Gf
+MNmP9CM5eovQOGx5ho8WqD+aTebs+k2tn92BBPqeZqpWRa5P/+jrdKml1qx4llHW
+MXrs3IgIb6+hUIB+S8dz8/mmO0bpr76RoZVCXYab2CZedFut7qc3WUH9+EUAH5mw
+vSeDCOUMYQR7R9LINYwouHIziqQYMAkGByqGSM44BAMDLwAwLAIUWXBlk40xTwSw
+7HX32MxXYruse9ACFBNGmdX2ZBrVNGrN9N2f6ROk0k9K
+-----END CERTIFICATE-----`;
+
+// This function would be a lot simpler if we used bluebird
+// to promisify fs, we just don't want to add a new bloated
+// dependency only for such small piece of code.
+//
+// fs module has the writeFile function, but tests showed that
+// it didn't flush data to disk before returning.
+function createTemporaryFile(data) {
+  const tmp = new temporary.File();
+  return new Promise((accept, reject) => {
+    fs.open(tmp.path, 'w', (err, fd) => {
+      err ? reject(err) : accept(fd);
+    });
+  }).then(fd => {
+    return new Promise((accept, reject) => {
+      fs.write(fd, data, err => {
+        err ? reject(err) : accept(fd);
+      });
+    });
+  }).then(fd => {
+    return new Promise((accept, reject) => {
+      fs.close(fd, err => {
+        err ? reject(err) : accept(tmp);
+      });
+    });
+  }).catch(err => tmp.unlik());
```

Comment:
`unlink`

Also, I think you want to re-reject the promise here?  Otherwise you'll just get back `undefined` from this promise.  `err => { tmp.unlink(); throw err; }`

Also, wouldn't https://nodejs.org/api/fs.html#fs_fs_writefilesync_file_data_options do the trick?